org.adempiere.base/src/org/compiere/util/DefaultEvaluatee.java

```diff
@@ -237,7 +237,9 @@ else if (m_tabNo <= 0)
 		
 		//try context if no data provider and not only window and not only tab
 		if (!globalVariable && m_dataProvider == null && Util.isEmpty(value) && !m_onlyWindow && (m_onlyTab == null || !m_onlyTab.booleanValue())) {
-			value = Env.getContext(ctx, variableName);
+			value = Env.getContext(ctx, m_windowNo, variableName);
```

Comment:
with m_windowNo should have been tried before reaching here.

Reply:
> with m_windowNo should have been tried before reaching here.

No, because GridFieldVO is parsing the context with this:
`vo.ColumnSQL = Env.parseContext(ctx, -1, vo.ColumnSQL, false, true);`
so, the windowNo is -1 and is not used above in the DefaultEvaluatee

Reply:
ok, will test, I'm completely lost on what's going on here ...

Reply:
Found out why, it is due to line 633 in Env.java where it auto add "#" to the context variable. In other word, `Env.getContext(Env.getCtx(), -1, "#AD_Client_ID", false)` and `Env.getContext(Env.getCtx(), -1, "AD_Client_ID", false)` will give you the same result. It is a special thing though in that method, as the same doesn't happens with `Env.getContext(Env.getCtx(), "#AD_Client_ID")` and `Env.getContext(Env.getCtx(), "AD_Client_ID")`.

Reply:
I think if we want to keep this behaviour for backward compatibility, instead of the changes above, we should duplicate the auto add "#" logic to the `Env.getContext(Env.getCtx(), "AD_Client_ID")` method.

Reply:
Propose alternative patch:
[env.globalcontext.patch.txt](https://github.com/user-attachments/files/18544710/env.globalcontext.patch.txt)

Reply:
Thanks @hengsin 

At first I was not convinced by your patch because it creates another backward incompatibility.

But after analyzing potential scenarios it seems that's harmless and your patch makes more consistent the two methods in the Env class.

The new test case passed correctly, and it solves the problem in our migration.

Approving this.